keps/sig-storage/20200127-recover-from-resize.md

```diff
@@ -0,0 +1,192 @@
+---
+title: Recover from volume expansion failure
+authors:
+  - "@gnufied"
+owning-sig: sig-storage
+participating-sigs:
+  - sig-api-machinery
+reviewers:
+  - "@msau42"
+  - "@liggit"
+approvers:
+  - "@saad-ali"
+editor: TBD
+creation-date: 2020-01-27
+last-updated: 2020-01-27
+status: implementable
+see-also:
+replaces:
+superseded-by:
+
+---
+
+# Recovery from volume expansion failure
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Implementation](#implementation)
+    - [User flow stories](#user-flow-stories)
+        - [Case 1 (controller+node expandable):](#case-1-controllernode-expandable)
+        - [Case 2 (controller+node expandable):](#case-2-controllernode-expandable)
+        - [Case 3](#case-3)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Graduation Criteria](#graduation-criteria)
+  - [Test Plan](#test-plan)
+  - [Monitoring](#monitoring)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Fix Quota when resize succeeds with reduced size](#fix-quota-when-resize-succeeds-with-reduced-size)
+  - [Allow admins to manually fix PVCs which are stuck in resizing condition](#allow-admins-to-manually-fix-pvcs-which-are-stuck-in-resizing-condition)
+- [Infrastructure Needed [optional]](#infrastructure-needed-optional)
```

Comment:
Please fill in the section: https://github.com/kubernetes/enhancements/tree/master/keps/NNNN-kep-template#production-readiness-review-questionnaire

Also the new template requires a kep to moved to a folder and have a README.md and kep.yaml.  See some examples here: https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture

Reply:
This KEP is using the existing enhancement issue for volume expansion, but it looks like this new format requires a separate enhancement issue for every KEP.

Reply:
created new issue and added PRR questions.